esmvalcore/_recipe/recipe.py

```diff
@@ -133,8 +133,8 @@ def _update_target_grid(dataset, datasets, settings):
     if dataset.facets['dataset'] == grid:
         del settings['regrid']
     elif any(grid == d.facets['dataset'] for d in datasets):
-        representative_ds = _representative_dataset(
-            _select_dataset(grid, datasets))
+        representative_ds = _representative_datasets(
+            _select_dataset(grid, datasets))[0]
```

Comment:
does that func always return an object of len >=1?

Reply:
I think so. The `[0]` part was previously included within `_representative_dataset`, so if it wasn't a problem then, it's not a problem now 😁